enrichments/trace/internal/elastic/span.go

```diff
@@ -300,6 +303,40 @@ func (s *spanEnrichmentContext) setServiceTarget(span ptrace.Span) {
 	}
 }
 
+func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
```

Comment:
[**For reviewers**] There is quite a bit of discrepancy between the destination resource and the service target fields (`service.target.{name, type}`) - you can check this by the test cases in the PR. The current code keeps the logic similar to what is present in apm-data for the most part --  any points of differences should be highlighted in the other comments.